### PR TITLE
Switch to GitHub Downloads

### PR DESCRIPTION
**Pull request by justinhhorner**
This PR changes the Godot download location to GitHub and adds tests that ensure the download URLs for export templates and each platform are generated correctly.

@definitelyokay Before merging, I'd appreciate any edge case scenarios you can think of for the install command that I might have missed. I'll adjust and tests to cover them.

### CODE REVIEW

**Comment by justinhhorner** [comment]
Ah, thank you! I missed that warning. Since we don't use the variable I removed it for now.

I just checked and unfortunately GitHub has the same file name discrepancies. I'd be happy to work on adding support for installing 3.x versions next.

File: GodotEnv/src/features/godot/models/GodotEnvironment.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ bool isTemplate
 public abstract class GodotEnvironment : IGodotEnvironment {
   public const string GODOT_FILENAME_PREFIX = "Godot_v";
   public const string GODOT_URL_PREFIX =
-    "https://downloads.tuxfamily.org/godotengine/";
+    "https://github.com/godotengine/godot/releases/download/";
```
Comment (garettbass) [inline GodotEnv/src/features/godot/models/GodotEnvironment.cs:118]: I think this URL should be:
```
https://github.com/godotengine/godot-builds/releases/download/
```
Because that is the prefix of the URL provided in the latest 4.2-beta1 announcement, and the URL you have above does not include 4.2-beta1.

Comment (garettbass) [reply GodotEnv/src/features/godot/models/GodotEnvironment.cs:118]: For more context, observe differences between these two pages:
[https://github.com/godotengine/godot-builds/releases/](https://github.com/godotengine/godot-builds/releases/)
- 4.2-beta1
- 4.1.2-stable
- 4.2-dev6
- etc.

[https://github.com/godotengine/godot/releases](https://github.com/godotengine/godot/releases)
- 4.1.2-stable
- 3.5.3-stable
- 4.0.4-stable
- etc.

It is particularly frustrating that there isn't a single URL that provides all of the releases, since I'm most interested in downloading 4.2-beta1.

Would it be a reasonable refactor to support querying both URLs to see whether either offers the requested version?

Comment (garettbass) [reply GodotEnv/src/features/godot/models/GodotEnvironment.cs:118]: Upon further review, it does look like `godot-builds/releases` may contain a strict superset of the releases available from `godot/releases`

**Comment by justinhhorner** [comment]
@garettbass Thanks for your feedback! I just pushed a commit to change the prefix URL. My local testing suggests this is all working as expected.

**Comment by justinhhorner** [comment]
@definitelyokay Yeah, good to merge 👍
